gatling-test/README.md

```diff
@@ -0,0 +1,72 @@
+=========================
+
+#Content
+
+This stress-test with Gatling will simulate the activity of 1000 users in the computer database webapp. It will allow you to discover some of your application's bottleneck.
+
+Two simulations are available, depending if you have already enabled Spring Security or not. If you've enabled Spring Security, the test will simulate the activity of 1000 normal users and 10 admins.
+
+The user simulation use the following scenario: authenticate, browse, and search for computers.
+
+The admin simulation use the following scenario: authenticate, browse, search, add, edition and deletion of a computer.
+
+
+#Installation
+You can copy the content of this folder in your existing maven project, or use it as a single project.
+
+#Configuration
+This test was created for a specific computer database application, but you can easily configure it to match yours. The configuration files are in the folder src/test/resources/.
+
+In application.conf, you must specify the URLs, forms params used in your application.
+
+The folder src/test/resources/data contains some csv feeders which are used in the test.
+* The file addComputer.csv contains values that are used when creating a new computer. The name is not present, he is created randomly.
+* The file admin.csv contains the username and login of an admin, and the file user.csv contains the username and login of a user.
+* The file search.csv contains a searchComputerName that should be present when a search with the searchCriterion is done.
+* The file searchColumn.csv contains the column parameter used in the get request for sorting the computers.
+* The file searchOrder.csv contains the order parameter used in the get request for sorting the computers.
+* The file searchPageSize.csv contains the page size parameter used in the get request for sorting the computers.
+
+You can add as many lines as you want in these csv feeders.
+
+If you want to modify the behavior of the test, you can modify the files in the folder src.scala.com.excilys.computerdatabase.gatling
+
+#Pre-requisites on your webapp
+
+For the test to work, some elements in your webapp must have a specific id:
+* In the dashboard, the checkbox used to choose which computer to delete must have the id: {name computer}_id
+* In the dashboard, the href used to edit a computer must have the id: {name_computer}_name
+
+#Utilisation
+
+If you have enabled Spring Security, type the command **mvn gatling:execute mvn gatling:execute -Dgatling.simulationClass=com.excilys.computerdatabase.gatling.simulation.ComputerDatabaseSimulationSecurity** to launch the test. You can view the results of the test in your browser, or in the folder target/gatling/results/
+
+If you have not yet enabled Spring Security, type the command **mvn gatling:execute mvn gatling:execute -Dgatling.simulationClass=com.excilys.computerdatabase.gatling.simulation.ComputerDatabaseSimulation** to launch the test.
+
+
+#Results
```

Comment:
Those are called KPI (Key Performance Indicators)

The scenario can be considered as performant and successful if all the following conditions are met:
 * Failed request count is 0
 * 95th percentile must be under 1200 ms

+ add a section "Scenario protocol" where you describe how many warmup runs have to be done (3 seems like a good one), and how many statistic runs have to be done (3 also seem like a good one).
The worst stats will be kept as result for KPIs.